### ISSUE TITLE
Control Plane Secret Errors

### ISSUE THREAD

**Issue by coreywagehoft** [kumahq/kuma#1354]
### Summary
I have Kuma installed in a multi-zone configuration.  I have been occasionally finding a remote control plane in a bad state with non-stop errors with failing to get secrets.  Here is an example error message:

```
2020-12-22T17:22:29.587Z    ERROR    xds-server.dataplane-sync-watchdog    OnTick() failed    {"dataplaneKey": {"Mesh":"default","Name":"<redacted dataplane name>"}, "error": "failed to list k8s Secrets: rpc error:code = Unavailable desc = transport is closing", "errorVerbose": "rpc error: code = Unavailable desc = transport is closing\nfailed to list k8s Secrets\ngithub.com/kumahq/kuma/pkg/plugins/secrets/k8s.(*KubernetesStore).List\n\t/go/src/github.com/kumahq/kuma/pkg/plugins/secrets/k8s/store.go:162\ngithub.com/kumahq/kuma/pkg/core/secrets/manager.(*secretManager).List\n\t/go/src/github.com/kumahq/kuma/pkg/core/secrets/manager/manager.go:50\ngithub.com/kumahq/kuma/pkg/core/resources/manager.(*customizableResourceManager).List\n\t/go/src/github.com/kumahq/kuma/pkg/core/resources/manager/customizable_manager.go:27\ngithub/com/kumahq/kuma/pkg/core/resources/manager.(*cachedManager).List\n\t/go/src/github.com/kumahq/kuma/pkg/core/resources/manager/cache.go:99\ngithub.com/kumahq/kuma/pkg/xds/cache/mesh.GetMeshSnapshot\n\t/go/src/github.com/kumahq/kuma/pkg/xds/cache/mesh/snapshot.go:60\ngithub.com/kumahq/kuma/pkg/xds/cache/mesh.(*Cache).GetHash.func1\n\t/go/src/github.com/kumahq/kuma/pkg/xds/cache/mesh/cache.go:67\ngithub.com/kumahq/kuma/pkg/xds/cache/once.(*Once).Do.func1\n\t/go/src/github.com/kumahq/kuma/pkg/xds/cache/once/once.go:13\nsync.(*Once).doSlow\n\t/usr/local/go/src/sync/once.go:66\nsync.(*Once).Do\n\t/usr/local/go/src/sync/once.go:57\ngithub.com/kumahq/kuma/pkg/xds/cache/once.(*Once).Do\n\t/go/src/github.com/kumahq/kuma/pkg/xds/cache/once/once.go:12\ngithub.com/kumahq/kuma/pkg/xds/cache/mesh.(*Cache).GetHash\n\t/go/src/github.com/kumahq/kuma/pkg/xds/cache/mesh/cache.go:64\ngithub.com/kumahq/kuma/pkg/xds/server.DefaultDataplaneSyncTracker.func1.2\n\t/go/src/github.com/kumahq/kuma/pkg/xds/server/components.go:266\ngithub.com/kumahq/kuma/pkg/util/watchdog.(*SimpleWatchdog).Start\n\t/go/src/github.com/kumahq/kuma/pkg/util/watchdog/watchdog.go:24\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1374"}
```

When the control plane gets into this state, pods fail to terminate and new pods are stuck in a pending state.  To fix this issue I delete the control plane pod, and when it restarts everything gets back into a working state.

### Steps To Reproduce

I have not been able to identify when this begins to occur.  I have not made any configuration updates in many days, and I just came across this when I noticed a pod taking a long time to terminate.

### Additional Details & Logs

- Version: 1.0.3
- Error logs: Log message is above
- Configuration: Multi-Zone
- Platform and Operating System: Kubernetes 1.14 on Azure Kubernetes Service

**Comment by subnetmarco** [comment]
Any thoughts @jakubdyszkiewicz @nickolaev ?

**Comment by nickolaev** [comment]
Not sure if it is related, but why K8s 1.14? According to AKS [docs](https://docs.microsoft.com/en-us/azure/aks/supported-kubernetes-versions), this is not even a supported version today.

**Comment by lobkovilya** [comment]
Hey @coreywagehoft, the error says when Kuma was trying to fetch Secrets from `kube-apiserver` it failed with error `transport is closing`. Is it possible that `kube-apiserver` experienced some troubles at those times, maybe it was restarting?

**Comment by coreywagehoft** [comment]
@lobkovilya it is entirely possible.  The strange part is this error would be repeating many times a second in the logs, and as soon as I deleted the pod, the new replica would not have the error.

It seems that when it had an issue talking with the k8s api-server it would get caught in an infinite loop with this error and never recover.

**Comment by nickolaev** [comment]
@coreywagehoft we are debugging, what we believe is a related issue. Do you by any chance have an idea if the CPU goes up once the `kuma-cp` gets into this infinite loop?

**Comment by nickolaev** [comment]
We have addressed a related issue, and I think it will solve that issue. @coreywagehoft can you please re-check if you still see theproblem.

**Comment by subnetmarco** [comment]
@coreywagehoft the fix is in v1.0.5
